crates/proof-of-sql/src/sql/proof_plans/union_exec_test.rs

```diff
@@ -0,0 +1,221 @@
+use super::test_utility::*;
+use crate::{
+    base::{
+        database::{
+            owned_table_utility::*, table_utility::*, ColumnType, OwnedTable,
+            OwnedTableTestAccessor, TableTestAccessor, TestAccessor,
+        },
+        map::indexmap,
+        scalar::Curve25519Scalar,
+    },
+    sql::{
+        proof::{
+            exercise_verification, FirstRoundBuilder, ProvableQueryResult, ProverEvaluate,
+            VerifiableQueryResult,
+        },
+        proof_exprs::test_utility::*,
+    },
+};
+use blitzar::proof::InnerProductProof;
+use bumpalo::Bump;
+
+#[test]
+fn we_can_prove_and_get_the_correct_empty_result_from_a_union_exec() {
+    let data0 = owned_table([bigint("a0", [0_i64; 0])]);
+    let t0 = "sxt.t0".parse().unwrap();
+    let data1 = owned_table([bigint("a1", [0_i64; 0])]);
+    let t1 = "sxt.t1".parse().unwrap();
+    let mut accessor = OwnedTableTestAccessor::<InnerProductProof>::new_empty_with_setup(());
+    accessor.add_table(t0, data0, 0);
+    accessor.add_table(t1, data1, 0);
+    let ast = union_exec(
+        vec![
+            projection(cols_expr_plan(t1, &["a1"], &accessor), tab(t1)),
+            projection(cols_expr_plan(t0, &["a0"], &accessor), tab(t0)),
+        ],
+        vec![column_field("a", ColumnType::BigInt)],
+    );
+    let verifiable_res = VerifiableQueryResult::new(&ast, &accessor, &());
+    exercise_verification(&verifiable_res, &ast, &accessor, t0);
+    let res = verifiable_res.verify(&ast, &accessor, &()).unwrap().table;
+    let expected_res = owned_table([bigint("a", [0_i64; 0])]);
+    assert_eq!(res, expected_res);
+}
+
+#[test]
+fn we_can_prove_and_get_the_correct_result_from_a_union_exec() {
+    let alloc = Bump::new();
+    let data0 = table([
+        borrowed_bigint("a0", [1_i64, 2, 3, 4, 5], &alloc),
+        borrowed_varchar("b0", ["1", "2", "3", "4", "5"], &alloc),
+    ]);
+    let t0 = "sxt.t0".parse().unwrap();
+    let data1 = table([
+        borrowed_bigint("a1", [2_i64, 3, 4, 5, 6], &alloc),
+        borrowed_varchar("b1", ["2", "3", "4", "5", "6"], &alloc),
+    ]);
+    let t1 = "sxt.t1".parse().unwrap();
+    let mut accessor = TableTestAccessor::<InnerProductProof>::new_empty_with_setup(());
```

Comment:
Can you use the NaiveEvaluationProof instead?

Reply:
Sure but that requires a different PR to refactor https://github.com/spaceandtimelabs/sxt-proof-of-sql/blob/main/crates/proof-of-sql/src/sql/proof/verifiable_query_result_test_utility.rs and replace every instance of `InnerProductProof` with `NaiveEvaluationProof` in examples that need to be modified.